changelog/next/features/4795--record-indexing.md

```diff
@@ -0,0 +1 @@
+Indexing records with string expressions is now supported.
```

Comment:
Can you provide a brief example in the changelog?

Reply:
We couldn't come up with a concise example that showed the expression evaluating to string and then accessing the record.